src/controller/offers.py

```diff
@@ -20,3 +20,38 @@ def __init__(self, **kwargs):
 def render_offers(request: Request): 
     offers = product_api.get_discounted_products()
     return templates.TemplateResponse("pages/offers.html", {"request": request, "offers": offers})
+
+@router.get("/favourites", response_class=HTMLResponse)
+def render_favourites(request: Request): 
+
+    if not check_token(request): 
+        return RedirectResponse(url="/login/", status_code=302)
+
+    laptops = product_api.get_top_products_by_type("portatil")
+    smartphones = product_api.get_top_products_by_type("smartphone")
+    monitors = product_api.get_top_products_by_type("monitor")
+
+    products_by_type = {
+        "Portátiles": laptops,
+        "Smartphones": smartphones,
+        "Monitores": monitors
+    }
+
+    category_urls = {
+        "Portátiles": "/category/laptops",
+        "Smartphones": "/category/smartphones",
+        "Monitores": "/category/monitors"
+    }
+
+    return templates.TemplateResponse("pages/index.html", {
+        "request": request,
+        "products_by_type": products_by_type,
+        "category_urls": category_urls
+    })
+
+def check_token(request: Request) -> bool:
```

Comment:
this function will you never use it again elsewhere?

Reply:
Done